utide/_time_conversion.py

```diff
@@ -2,44 +2,79 @@
 Utility for allowing flexible time input.
 """
 
-import warnings
+import numpy as np
 
-from datetime import date, datetime
 
+# to be added to get 1 on 1st January of year 1 from unix epoch '1970-01-01'
+_DAY_TO_GREGORIAN_EPOCH = 719163
 
-try:
-    from datetime import timezone
+# milisecond in a day
+_MS_PER_DAY = 1000 * 86400
 
-    have_tz = True
-except ImportError:
-    have_tz = False
+# datenum types
+_DATENUM_TYPES = (np.float, np.int64, np.int32)
 
 
-def _normalize_time(t, epoch):
-    if epoch == "python":
-        return t
-    if epoch == "matlab":
-        return t - 366
+def _date2num(date, epoch="1970-01-01 00:00:00.000"):
+    """
+    Numpy based date to datenum calculator.
+
+    `date` can be anything parsable by np.datetime64 - string, datetime, datetime64,
+    pandas datetime.
+
+    `epoch` is taken as the unix epoch.
+    """
+    date = np.atleast_1d(date)
```

Comment:
Is there a reason for forcing this to have at least one dimension rather than using `asarray`?

Reply:
Thanks for your review. 

Not really. It is probably not needed either as the same function has been applied over `t` at various other places in the codebase. For me, it is a short-hand way to change the argument float/int/array to a NumPy array, while asking it to have at least 1d length. We can change it to asarray safely I think.

Reply:
OK, please try switching to asarray.